lib/test.js

```diff
@@ -22,68 +22,39 @@ const captureStack = start => {
 	return obj.stack;
 };
 
-const assertions = assert.wrapAssertions({
-	pass(test) {
-		test.countPassedAssertion();
-	},
-
-	pending(test, promise) {
-		test.addPendingAssertion(promise);
-	},
-
-	fail(test, error) {
-		test.addFailedAssertion(error);
-	}
-});
-const assertionNames = Object.keys(assertions);
-
-function log(...inputArgs) {
-	const args = inputArgs.map(value => {
-		return typeof value === 'string' ?
-			value :
-			concordance.format(value, concordanceOptions);
-	});
-
-	if (args.length > 0) {
-		this.addLog(args.join(' '));
-	}
-}
-
-function plan(count) {
-	this.plan(count, captureStack(this.plan));
-}
-
-function timeout(ms) {
-	this.timeout(ms);
-}
-
 const testMap = new WeakMap();
-class ExecutionContext {
+class ExecutionContext extends assert.Assertions {
 	constructor(test) {
+		super();
 		testMap.set(this, test);
 
-		const skip = () => {
-			test.countPassedAssertion();
-		};
-
-		const boundPlan = plan.bind(test);
-		boundPlan.skip = () => {};
-
-		Object.defineProperties(this, assertionNames.reduce((props, name) => {
-			props[name] = {value: assertions[name].bind(test)};
-			props[name].value.skip = skip;
-			return props;
-		}, {
-			log: {value: log.bind(test)},
-			plan: {value: boundPlan},
-			timeout: {value: timeout.bind(test)}
```

Comment:
Doesn't look like these are still bound?

Reply:
I'm not sure if they need to be bound to test. These functions now exist as part of the ExecutionContext and are bound to ExecutionContext, similarly to all other assertions. If they need to access the test instance, they can do so via WeakMap. 

Is it acceptable behaviour or would you rather keep those functions bound to the Test instance?

Reply:
By bound, I mean that this should work:

```js
const { log } = t
someArray.forEach(log)
```

Looks like we don't have test coverage for this but that should keep working.

Reply:
Oh, I see.

I was going to add the test which verifies that these functions are properly bound, but then I noticed that this test already exists - https://github.com/avajs/ava/blob/master/test/test.js#L685 
I could add another more explicit test, if you would like?

Reply:
Turns out,
```js
const { log } = t
someArray.forEach(log)
```
does not work

Reply:
Fixed by moving into constructor.

Another solution would be to implement log, plan and timeout as getters (like end and title), because those seem to be binding to instance properly. However that way it seemed weird, I decided to put into constructor.